tests/util/test_async_helpers.py

```diff
@@ -313,13 +318,22 @@ async def caller():
         self.successResultOf(d2)
 
 
-class StopCancellationTests(TestCase):
-    """Tests for the `stop_cancellation` function."""
+@parameterized_class(
+    ("wrap_deferred",),
+    [
+        (lambda _self, deferred: stop_cancellation(deferred),),
+        (lambda _self, deferred: delay_cancellation(deferred, all=True),),
+    ],
+)
```

Comment:
This is rather ugly. Alternatives are welcome.

I previously tried an abstract base class, but trial tried to instantiate it.

Reply:
>  but trial tried to instantiate it.

did you try giving it a different name? AIUI trial picks the things to instantiate based on their name. (IIRC it's things ending in `Test`, `Tests` or `TestCase`.)

Reply:
(otherwise: rather than using a lambda, just have a boolean, and make `wrap_deferred` a real method that inspects the boolean before delegating to `stop_cancellation`/`delay_cancellation`. It's marginally less ugly.)

Reply:
> try giving it a different name

Renaming the class doesn't work. trial seems to look for subclasses of `TestCase` rather than by matching on names.

I'm not a fan of using booleans as enums, so I'll switch it to a string. Which is still not an enum but at least makes the meaning clear.